### PR TITLE
Add product's Azure Vault to terraform

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-302

### Change description ###

Add Azure Vault for the consumer (the same instance as used by the producer, as it's product-level). Consumer terraform needs this vault to pass connection details to smoke/functional tests running on Jenkins box.

Successful test build:
https://sandbox-build.platform.hmcts.net/job/HMCTS/job/send-letter-consumer-service/job/cnp/28/

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -64,3 +64,14 @@ module "consumer" {
...
 }
+
+module "key-vault" {
+  source              = "git@github.com:contino/moj-module-key-vault?ref=master"
+  product             = "${var.product}"
+  env                 = "${var.env}"
+  tenant_id           = "${var.tenant_id}"
+  object_id           = "${var.jenkins_AAD_objectId}"
+  resource_group_name = "${var.product}-producer-${var.env}"
+  # dcd_cc-dev group object ID
```
Comment (nitinprabhu) [inline infrastructure/main.tf:75]: could remove this comment ?

**Review by nitinprabhu — APPROVED** [review]
✈️

File: infrastructure/main.tf [hunk 2]
```diff
@@ -64,3 +64,14 @@ module "consumer" {
...
   }
 }
+
+module "key-vault" {
+  source              = "git@github.com:contino/moj-module-key-vault?ref=master"
+  product             = "${var.product}"
+  env                 = "${var.env}"
+  tenant_id           = "${var.tenant_id}"
+  object_id           = "${var.jenkins_AAD_objectId}"
+  resource_group_name = "${var.product}-producer-${var.env}"
```
Comment (lgonczar) [inline infrastructure/main.tf:74]: Unfortunately, the existing vault is named `send-letter-{env}`, but it's located in `send-letter-producer-{env}` resource group.

File: infrastructure/main.tf (hunk 1 above)
Comment (lgonczar) [reply infrastructure/main.tf:75]: I don't know - maybe it's better to know what group is related with that identifier?

**Pull request merged**